File: part-02/src/main/java/eu/luminis/workshop/smallsteps/persistence/InMemoryLegoPartCatalog.java
package eu.luminis.workshop.smallsteps.persistence;

import eu.luminis.workshop.smallsteps.logic.domainmodel.valueobjects.LegoSetNumber;
import eu.luminis.workshop.smallsteps.logic.domainservice.catalog.LegoPartCatalog;
import org.springframework.stereotype.Service;

import java.util.Collections;
import java.util.Map;

@Service
public class InMemoryLegoPartCatalog implements LegoPartCatalog {
    private final Map<LegoSetNumber, Map<String, Integer>> partsForLegoSets;

    public InMemoryLegoPartCatalog() {
        this(null);
    }

    public InMemoryLegoPartCatalog(Map<LegoSetNumber, Map<String, Integer>> partsForLegoSets) {
        this.partsForLegoSets = partsForLegoSets;
    }

    @Override
    public Map<String, Integer> allPartsForLegoSet(LegoSetNumber legoSetNumber) {
        return partsForLegoSets.getOrDefault(legoSetNumber, Collections.emptyMap());
    }
}

File: part-02/src/main/kotlin/eu/luminis/workshop/smallsteps/persistence/InMemoryLegoPartCatalog.kt
package eu.luminis.workshop.smallsteps.persistence

import eu.luminis.workshop.smallsteps.logic.domainModel.valueObjects.LegoSetNumber
import eu.luminis.workshop.smallsteps.logic.domainService.catalog.LegoPartCatalog

class InMemoryLegoPartCatalog(private val partsForLegoSets: LegoPartLookup) : LegoPartCatalog {
    override fun allPartsForLegoSet(legoSet: LegoSetNumber): Map<String, Int>? = partsForLegoSets[legoSet]
}

typealias LegoPartLookup = Map<LegoSetNumber, Map<String, Int>>